source_files/edge/p_setup.cc

```diff
@@ -1990,161 +1777,102 @@ static void LoadUDMFSideDefs(parser_t *psr)
 			int flags = 0, v1 = 0, v2 = 0;
 			int side0 = -1, side1 = -1, tag = -1;
 			int special = 0;
-
-			// process lindef block
-			while (1)
+			for (;;)
 			{
-				if (!GetNextAssign(psr, (uint8_t*)ident, (uint8_t*)val))
-				{
-					uint8_t *lp = psr->line;
-					while (*lp != 0 && *lp != 0x7D)
-						lp++; // find end of line or '}'
-					if (*lp == 0x7D)
-					{
-						break; // end of block
-					}
-					if (psr->next >= psr->length)
-					{
-						break; // end of lump
-					}
+				if (lex.Match("}"))
+					break;
 
-					continue; // skip line
-				}
-				// process assignment
-				if (epi::case_cmp(ident, "id") == 0)
-				{
-					tag = str2int(val, -1);
-				}
-				else if (epi::case_cmp(ident, "v1") == 0)
-				{
-					v1 = str2int(val, 0);
-				}
-				else if (epi::case_cmp(ident, "v2") == 0)
-				{
-					v2 = str2int(val, 0);
-				}
-				else if (epi::case_cmp(ident, "special") == 0)
-				{
-					special = str2int(val, 0);
-				}
-				else if (epi::case_cmp(ident, "sidefront") == 0)
-				{
-					side0 = str2int(val, -1);
-				}
-				else if (epi::case_cmp(ident, "sideback") == 0)
-				{
-					side1 = str2int(val, -1);
-				}
-				else if (epi::case_cmp(ident, "blocking") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0001;
-				}
-				else if (epi::case_cmp(ident, "blockmonsters") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0002;
-				}
-				else if (epi::case_cmp(ident, "twosided") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0004;
-				}
-				else if (epi::case_cmp(ident, "dontpegtop") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0008;
-				}
-				else if (epi::case_cmp(ident, "dontpegbottom") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0010;
-				}
-				else if (epi::case_cmp(ident, "secret") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0020;
-				}
-				else if (epi::case_cmp(ident, "blocksound") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0040;
-				}
-				else if (epi::case_cmp(ident, "dontdraw") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0080;
-				}
-				else if (epi::case_cmp(ident, "mapped") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0100;
-				}
-				else if (epi::case_cmp(ident, "passuse") == 0)
-				{
-					if (str2bool(val))
-						flags |= 0x0200; // BOOM flag
-				}
+				std::string key;
+				std::string value;
+				epi::token_kind_e block_tok = lex.Next(key);
+
+				if (block_tok == epi::TOK_EOF)
+					I_Error("Malformed TEXTMAP lump: unclosed block\n");
+
+				if (block_tok != epi::TOK_Ident)
+					I_Error("Malformed TEXTMAP lump: missing key\n");
+
+				if (! lex.Match("="))
+					I_Error("Malformed TEXTMAP lump: missing '='\n");
+
+				block_tok = lex.Next(value);
+
+				if (block_tok == epi::TOK_EOF || block_tok == epi::TOK_ERROR || value == "}")
+					I_Error("Malformed TEXTMAP lump: missing value\n");
+
+				if (! lex.Match(";"))
+					I_Error("Malformed TEXTMAP lump: missing ';'\n");
+
+				if (key == "id")
```

Comment:
Probably a good idea on the else if here, this is a lot of string compares.  They should be ordered by commonality, I would also guess it would make sense to generate id's for the keys, so can do comparisons that way.